coursebuilder/common/utils.py

```diff
@@ -339,3 +340,22 @@ def find_youtube_video_id(text):
         if match:
             return match.group(1)
     return None
+
+
+def embed_carousel(content):
+    start_key = '[start_carousel]'
+    stop_key = '[stop_carousel]'
+    check = True
+
+    while check:
+        start = content.find(start_key)
```

Comment:
@andela-angene Does this only assumes presence of one start key and that we can't have start -> start -> end. i.e a block that was not closed.

Reply:
The function only picks sets of open and closed keywords. For start->start->end, only one carousel will be rendered. for start->end start->end, the two carousels will be rendered.

Reply:
@andela-angene The start->start->end while still a user error is bound to happen let's check that they are even numbered.

Reply:
@andela-angene Case in point https://gmkr.io/s/59c216a0a5d5ba6b41656ec6/0

Reply:
@chegejames am already checking this, only pairs of start->end keys will render. If the user opens the carousel and then starts pasting in content, there no way to predict where he/she intended for the carousel to stop within the content. I will update the instruction to make this clear, what are your thoughts on this?

Reply:
@andela-angene Please have a link to code when quoting lines or blocks e.g https://github.com/andela-angene/coursebuilder-core/blob/7e629f6bac182e1157e142d3664bfe0b7369c788/coursebuilder/common/utils.py#L353

Reply:
@andela-angene You are checking for consecutive start and end keys e.g '[start_carousel] 1 1 [split] [start_carousel] 2 1 [split] 2 2 [stop_carousel] [split] 1 2 [stop_carousel]' would creates a recursive carousel. We need a more robust parser and that preview comes in very handy here so that the user can see what they are getting.

Reply:
@andela-angene The recursion https://github.com/andela-angene/coursebuilder-core/blob/7e629f6bac182e1157e142d3664bfe0b7369c788/coursebuilder/common/utils.py#L350 is greedy. In the first iteration of the example above content='<div class="owl-carousel owl-theme course-carousel"><div class="item"><p>1 1 </p></div><div class="item"><p>[start_carousel] 2 1 </p></div><div class="item"><p>2 2 </p></div></div> [split] 1 2 [stop_carousel]'

Reply:
Updated.